### PR TITLE
[asl] Division improvements

### PR DESCRIPTION
**Pull request by HadrienRenaud**
This PR implements various features around basic integer division.

- [asl] Implement more division operations
- [asl] Implement first Division by zero type-check
- [asl] Support literal operations as static
- [asl] Fix build of bitvectors from integers

### CODE REVIEW

File: asllib/bitvector.ml [hunk 1]
```diff
@@ -297,6 +282,9 @@ let of_z sz z =
   do_rec msk (n - 1);
   (sz, Bytes.unsafe_to_string result)
 
+let of_int_sized sz i = of_z sz (Z.of_int i)
+let _of_int i = of_int_sized (Sys.int_size - 1) i
```
Comment (maranget) [inline asllib/bitvector.ml:286]: Isn't function `_of_int` dead code?

**Comment by maranget** [comment]
Hi @HadrienRenaud, LGTM up to a minor remark. Would you please rebase on master and squash?

File: asllib/bitvector.ml (hunk 1 above)
Comment (HadrienRenaud) [reply asllib/bitvector.ml:286]: I think it is. Will delete.

**Comment by HadrienRenaud** [comment]
@maranget Done. Could you merge?

**Comment by maranget** [comment]
Merged, thanks @HadrienRenaud.

**Pull request merged**